ukis_pysat/download.py

```diff
@@ -148,117 +127,78 @@ def query_metadata(self, platform, date, aoi, cloud_cover=None):
                 )
 
         try:
-            # construct list of harmonized metadata
+            # construct MetadataCollection from list of Metadata objects
             meta = []
             for m in meta_src:
                 meta.append(self.construct_metadata(meta_src=m))
-            return meta
+            return MetadataCollection(meta)
         except Exception as e:

```

Comment:
We have many of these general try, except blocks. What was the purpose of them again? Wouldn't it be better to just get rid of them?

Reply:
Good point. Now is a good time to revise them again and remove what is not strictly needed.